File: src/test/java/it/gabrieletondi/telldontaskkata/doubles/TestOrderRepository.java
package it.gabrieletondi.telldontaskkata.doubles;

import it.gabrieletondi.telldontaskkata.domain.Order;
import it.gabrieletondi.telldontaskkata.repository.OrderRepository;

public class TestOrderRepository implements OrderRepository {
    private Order insertedOrder;

    public Order insertedOrder() {
        return insertedOrder;
    }

    public void save(Order order) {
        this.insertedOrder = order;
    }
}

File: kotlin/src/test/java/it/gabrieletondi/telldontaskkata/doubles/TestOrderRepository.kt
package it.gabrieletondi.telldontaskkata.doubles

import it.gabrieletondi.telldontaskkata.domain.Order
import it.gabrieletondi.telldontaskkata.repository.OrderRepository
import java.util.*

class TestOrderRepository : OrderRepository {
    var savedOrder: Order? = null
        private set
    private val orders: MutableList<Order> = ArrayList()
    override fun save(order: Order) {
        savedOrder = order
    }

    override fun getById(orderId: Int): Order {
        return orders.stream().filter { o: Order -> o.id == orderId }.findFirst().get()
    }

    fun addOrder(order: Order) {
        orders.add(order)
    }
}